### PR TITLE
feat: consolidate field path handling and enhance resolver

### PR DESCRIPTION
**Pull request by a-hilaly**
Our codebase had several different implementations for parsing and building path
segments used to locate `CEL` expressions for replacement. This resulted in
duplicated logic, inconsistent handling of special cases, and made it harder to
maintain and extend path related functionalities. This patch consolidates all path
handling into a dedicated package and standardizes the approach across the
codebase.

This patch also includes a partial reimplementation of the resolver with
comprehensive unit tests to ensure reliable handling of complex path scenarios and
edge cases. The new implementation provides better error reporting and more
predictable behavior when dealing with nested structures and arrays.

Key changes:
- Add new fieldpath package for consistent parsing and building of field paths
- Standardize path formats to use dot notation and quoted fields for special characters
- Replace custom path parsing with a more robust implementation
- Update path handling in builder.go and resolver.go to use new fieldpath package
- Fix path prefix handling by removing '>' character usage
- Add comprehensive tests for path parsing and building
- Improve array handling in path resolution and value setting
- Add extensive test coverage for resolver edge cases

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: internal/typesystem/parser/parser.go [hunk 1]
```diff
@@ -223,3 +224,16 @@ func isInteger(v interface{}) bool {
...
+// .fieldName to avoid ambiguity and simplify parsing back the path.
+func joinPathAndFieldName(path, fieldName string) string {
+	if fieldName == "" || strings.Contains(fieldName, ".") {
+		return fmt.Sprintf("%s[%q]", path, fieldName)
+	}
+	if path == "" {
+		return fieldName
+	}
+	return fmt.Sprintf("%s.%s", path, fieldName)
+}
```
Comment (a-hilaly) [inline internal/typesystem/parser/parser.go:239]: I'm thinking about moving this to `fieldpath` package - eventually

File: internal/typesystem/resolver/resolver.go [hunk 2]
```diff
@@ -195,160 +190,115 @@ func (r *Resolver) getValueFromPath(path string) (interface{}, error) {
 
 // setValueAtPath sets a value in the resource using a dot-separated path.
 func (r *Resolver) setValueAtPath(path string, value interface{}) error {
```
Comment (a-hilaly) [inline internal/typesystem/resolver/resolver.go:192]: This one was a painful one to get right, the previous implementation didn't really work, we missed a few edge cases.

File: internal/typesystem/parser/parser.go [hunk 3]
```diff
@@ -223,3 +224,16 @@ func isInteger(v interface{}) bool {
 		return false
 	}
 }
+
+// joinPathAndField appends a field name to a path. If the fieldName contains
+// a dot or is empty, the path will be appended using ["fieldName"] instead of
+// .fieldName to avoid ambiguity and simplify parsing back the path.
+func joinPathAndFieldName(path, fieldName string) string {
+	if fieldName == "" || strings.Contains(fieldName, ".") {
+		return fmt.Sprintf("%s[%q]", path, fieldName)
```
Comment (michaelhtm) [inline internal/typesystem/parser/parser.go:233]: I was not aware of `%q` 🔥
